libs/designsystem/header-actions/src/header-actions.component.ts

```diff
@@ -44,19 +44,28 @@ export class HeaderActionsComponent implements AfterViewInit {
   @ViewChild('hiddenLayer', { read: ElementRef }) hiddenLayer!: ElementRef<HTMLElement>;
   @ViewChild('visibleLayer', { read: ElementRef }) visibleLayer!: ElementRef<HTMLElement>;
 
+  @HostBinding('class.is-collapsed')
+  _isCollapsed: boolean = true;
   _collapsedActions: string[] = [];
   _visibleActions: number;
 
   private dropdownTextToButtonMap: Map<string, HTMLButtonElement>;
+  private hiddenButtonsIntersectionObserver: IntersectionObserver;
+  private intersectionObserverOptions;
 
-  constructor(private renderer: Renderer2) {}
+  constructor(private renderer: Renderer2, private elementRef: ElementRef) {}
 
   ngAfterViewInit(): void {
-    this.initializeCollapsing();
+    if (this._visibleActions) {
+      this.showVisibleActions();
+    } else {
+      setTimeout(() => {
```

Comment:
The need for this timeout concerns me a bit :/ 

I am getting wrong intersection notifications on initialization of the component, if I dont wait for a moment before starting my observations.

Seems like everything 'has to settle'.